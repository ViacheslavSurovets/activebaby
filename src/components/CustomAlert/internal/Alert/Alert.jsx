import React from 'react';
import { connect } from 'react-redux';
import { AlertContainer, AlertText } from '../../styles';
import { createStructuredSelector } from 'reselect';
import { selectShowAlert } from '@redux/alert/alert.selectors';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import { CheckedIcon, FailIcon } from '@icons';

const Alert = ( { alert } ) => {
  const { t } = useTranslation ();
  return (
    <>
      {
        alert.map ( alert =>
          <AlertContainer key={ alert.id }>
            <AlertText>{ t ( alert.msg ) }</AlertText>
            {
              alert.type === 'success' ?
                <CheckedIcon color='#00ff59' />
                :
                <FailIcon color='#fa1005' /> }
          </AlertContainer>
        )
      }
    </>
  );
};

const mapStateToProps = createStructuredSelector ( {
  alert: selectShowAlert
} );

export default connect ( mapStateToProps ) ( Alert );

Alert.propTypes = {
  alert: PropTypes.array
};
