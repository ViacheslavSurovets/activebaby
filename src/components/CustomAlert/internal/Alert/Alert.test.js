import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import Alert from './Alert';
import configureMockStore  from 'redux-mock-store';


const mockStore = configureMockStore();
const store = mockStore({});

describe('Alert', ()=> {
  it('Alert Snapshot', ()=> {
    expect ( shallow (
      <Provider store={store}>
        <Alert  />
      </Provider>
    ) ).toMatchSnapshot();
  });
});
