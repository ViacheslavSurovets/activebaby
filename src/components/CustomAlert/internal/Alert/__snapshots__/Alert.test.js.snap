// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Alert Snapshot 1`] = `ShallowWrapper {}`;
